templates/testutil/prompt/prompt.go

```diff
@@ -131,6 +136,71 @@ func DialogTest(ctx context.Context, tb testing.TB, steps []DialogStep, cmd cli.
 	return nil
 }
 
+type FakePrompter struct {
+	steps     []DialogStep // Immutable.
+	stepIndex int          // Mutable. Incremented after each back-and-forth
+}
+
+func (f *FakePrompter) Prompt(ctx context.Context, msg string, args ...any) (string, error) {
+	// Note for maintainers: we call panic() rather than t.Fatal() in this
+	// function because panic() will stop all goroutines while t.Fatal() won't.
+	// If the other goroutines are left running, they will produce misleading
+	// failure messages that will obscure the true cause of the problem.
```

Comment:
Maybe worth clarifying what the _other_ goroutines are? Eyeballing this PR, we only spawn one goroutine in `DialogTestFunc` per each subtest. I wonder if `t.Fatal` + return an error would just work?

Using `panic` would terminate other tests, wouldn't it?

Reply:
This message is written from the point of view of the `prompt` library; it doesn't really know who's calling or what goroutines may be running, but it's safe to assume that one or more other goroutines are doing a test and are will soon do assertions that we hope to avoid (like timing out).

Yeah, `panic` will disrupt other tests, which kinda sucks. But I think it's worth it to achieve clarity of failure messages in the rare event of a test failure.